### ISSUE TITLE
Possible envoy optimizations

### ISSUE THREAD

**Issue by vincentjames501** [NickMRamirez/Proxy-Benchmarks#6]
I'm 100% not an envoy expert, I'm wondering if any of the following could help boost performance though:

```
type: LOGICAL_DNS
```

Could this just be `static`?

`server_header_transformation`

I think nginx and haproxy just pass headers through. Envoy will lowercase and normalize them all. This may be worth disabling.

`generate_request_id`

Since generating uuidv4's can be expensive per their documentation, may be worth disabling it to compare apples to apples.

May be interesting to play with the `—concurrency` option as well.

There are a few other params that can do less processing on the requests as well. May be worth looking at the actual request headers the application receives and the response headers the client receives as it may be transferring a lot more data around.

**Comment by NickMRamirez** [comment]
It looks like @daniel-corbett commit had changed the cluster service discovery type from "STATIC" to "LOGICAL_DNS". I will revert that because it works with STATIC and I do want to use IP addresses and not DNS for all benchmarks.

However, I don't agree that I should try to optimize Envoy's configuration regarding server header transformation or generating IDs. This benchmarks project is based off of Loggly's blog post "Benchmarking 5 Popular Load Balancers", wherein they say:

> The intent of these particular benchmarks is to show out-of-the-box configuration profiles without optimization, and outside of having a backend to another service, use the load balancer’s default configuration.

Show me Envoy documentation that shows that those settings you mentioned are what a suggested default or "getting started" configuration should be and I will make those changes. But in general, once we go down the road of optimizations, then I would need to do research about the other load balancers to see how to optimize them too. I would no doubt get it wrong, since I am not an expert on all of these optimizations, so it's best avoided altogether.

**Comment by daniel-corbett** [comment]
For reference, I updated the config to be based on the Envoy v3 spec as previously it was v2. It was quite difficult to determine what exactly needed to change between v2 and v3 to make things work so I copied over the default config from the [Docker container](https://www.envoyproxy.io/docs/envoy/latest/start/docker) and changed the options required to make it work in this benchmark environment.

> I think nginx and haproxy just pass headers through. Envoy will lowercase and normalize them all. This may be worth disabling.

Since [HAProxy 2.0](https://www.haproxy.com/blog/haproxy-2-0-and-beyond/#http-representation-htx), HTX is enabled out-of-the-box which also lower cases headers.
